README.md

```diff
@@ -1,47 +1,271 @@
+maximum number of hops needed to do lookups.
 
-## Algorithm
 
-The algorithm works by attempting to create a tree of deltas, produced by
+## Compression Algorithm
+
+The algorithm works by attempting to create a *tree* of deltas, produced by
 appending state groups to different "levels". Each level has a maximum size, where
-each state group is appended to the lowest level that is not full.
+each state group is appended to the lowest level that is not full. This tool calls a 
+state group "compressed" once it has been added to
+one of these levels.
 
 This produces a graph that looks approximately like the following, in the case
 of having two levels with the bottom level (L1) having a maximum size of 3:
 
 ```
 L2 <-------------------- L2 <---------- ...
 ^--- L1 <--- L1 <--- L1  ^--- L1 <--- L1 <--- L1
-```
 
-The sizes and number of levels used can be controlled via `-l`, and defaults to 3
-levels of sizes 100, 50 and 25.
+NOTE: A <--- B means that state group B's predecessor is A
+```
+The structure that synapse creates by default would be equivalent to having one level with
+a maximum length of 100. 
 
 **Note**: Increasing the sum of the sizes of levels will increase the time it
-takes for to query the full state of a given state group. By default Synapse
-attempts to keep this below 100.
+takes to query the full state of a given state group.
+
+# Automated tool: auto_compressor
+
+## Introduction:
+
+This tool is significantly more simple to use than the manual tool (described below).
+It scans through all of the rows in the `state_groups` database table from the start. When
+it finds a group that hasn't been compressed, it runs the compressor for a while on that
+group's room, saving where it got up to. After compressing a number of these chunks it stops,
+saving where it got up to for the next run of the `auto_compressor`.
+
+It creates three extra tables in the database: `state_compressor_state` which stores the
+information needed to stop and start the compressor for each room, `state_compressor_progress`
+which stores the most recently compressed state group for each room and `state_compressor_total_progress`
+which stores how far through the `state_groups` table the compressor has scanned.
+
+The tool can be run manually when you are running out of space, or be scheduled to run
+periodically.
+
+The output from the auto_compressor will be sent to `auto_compressor.log` (in the directory
+that the compressor is run from).
+
+## Building 
+
+This tool requires `cargo` to be installed. See https://www.rust-lang.org/tools/install
+for instructions on how to do this.
+
+To build `auto_compressor`, clone this repository and navigate to the `autocompressor/` 
+subdirectory. Then execute `cargo build`.
+
+This will create an executable and store it in `auto_compressor/target/debug/auto_compressor`.
+
+## Example usage
+```
+$ auto_compressor -p postgresql://user:pass@localhost/synapse -c 500 -l '100,50,25' -n 100
```

Comment:
```suggestion
$ auto_compressor -p postgresql://user:pass@localhost/synapse
```

I think we should not encourage people to change these parameters, since its really hard to know what the right values should be

Reply:
I've removed the specifying level sizes from the example, but I think it's helpful to keep the chunk size and number of chunks to be in the example to demonstrate some 'reasonable' parameters